server/server.go

```diff
@@ -2,41 +2,149 @@ package chatbackend
 	// Create new gorillaRouter
 	gorillaRouter := gorillaMux.NewRouter()
 	// Bind an endpoint path to handleWebSocketConnection
 	gorillaRouter.HandleFunc("/chat", HandleWebSocketConnection)
+	gorillaRouter.HandleFunc("/profile", HandleProfile)
 	// Listen on a specific port for incoming connections
 	if listenAndServeErr := http.ListenAndServe(":8080", gorillaRouter); listenAndServeErr != nil {
 		// If there is an error while setting up, panic and show us the error
 		log.Fatal("ListenAndServe: " + listenAndServeErr.Error())
 	} // if listenAndServeErr
 } // func startWebSocketServer
 
+func getProfileFromStorage(identityPublicKeyHex string) ([]byte, error) {
+	// Create a new connection to the firestore storage service
+	firestoreConnection, firestoreConnectionErr := newFirestoreConnection()
+	if firestoreConnectionErr != nil {
+		return nil, firestoreConnectionErr
+	} // if firestoreClientErr != nil
+	// Initialise an empty context with no values, no deadline, which will never be canceled
+	networkContext := context.Background()
+	// Get a snapshot of the document that contains the data we are interested in
+	documentSnapshot, documentSnapshotErr := firestoreConnection.Collection("clients").Doc(identityPublicKeyHex).Get(networkContext)
+	// If there is an error fetching the document
+	if documentSnapshotErr != nil {
+		// Log the error but don't return it as it may leak non-sensitive data that it's better not to be leaked
+		log.Println(documentSnapshotErr)
+		return nil, errors.New("Profile not found")
+	} // if documentSnapshotErr != nil
+	// Retreive the data from the document snapshot into a map[string]interface{}
+	clientDataMap := documentSnapshot.Data()
+	// If the client profile exists in the storage
+	if singleUserProfileBase64, exists := clientDataMap["profile"]; exists {
+		// Base64 decode the it to get the protobuf bytes
+		singleUserProfile, singleUserProfileErr := base64.StdEncoding.DecodeString(singleUserProfileBase64.(string))
+		// If there is an error with the base64 decoding, return the error
+		if singleUserProfileErr != nil {
+			return nil, singleUserProfileErr
+		} // if singleUserProfileErr
+		return singleUserProfile, nil
+	} // if singleUserProfileBase64, exists
+	return nil, errors.New("Profile not found")
+}
+
+func persistProfileToStorage(identityPublicKeyHex string, profileBase64 string) error {
+	// Create a new connection to the firestore storage service
+	firestoreConnection, firestoreConnectionErr := newFirestoreConnection()
+	if firestoreConnectionErr != nil {
+		return firestoreConnectionErr
+	} // if firestoreConnectionErr != nil
+	// Persist the client profile
+	_, firestoreWriteDataErr := firestoreConnection.Collection("clients").Doc(identityPublicKeyHex).Set(context.Background(), map[string]interface{}{
```

Comment:
before we persist we need to verify the integrity of the profile data -> you have to decode it and verify the signature. The `profile` package provides your the utils you might need for this.

Reply:
Added.